linera-service/src/proxy.rs

```diff
@@ -37,6 +38,12 @@ pub struct ProxyOptions {
     /// Timeout for receiving responses (us)
     #[structopt(long, default_value = "4000000")]
     recv_timeout_us: u64,
+
+    /// Use this when running from a Kubernetes cluster (including from within GCP)
+    /// Won't use config files to determine host IPs, but will do it dynamically
+    /// based on cluster info
+    #[structopt(long)]
+    kube: bool,
```

Comment:
Hm, I have a feeling like a feature gate is more appropriate here than a boolean flag. Furthermore the featuregate will conditionally compile the kube dependencies only when they are specified explicitly.

Reply:
This might be basic and I failed to find a reference to it, but how can I conditionally fetch the IPs from kubernetes if it becomes a feature gate?

Reply:
Good idea to introduce a cargo feature to control dependencies:
* Can we use the cargo feature `kube` to control whether the dynamic option `--kube` is available?
* Can we make sure that the service behaves as usual when `--kube` is absent, even if the feature is present.